Managers/LuaMan.cpp

```diff
@@ -522,6 +533,90 @@ namespace RTE {
 		return false;
 	}
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	bool LuaMan::SaveScriptedScene(const std::string &fileName) {
+		Scene* scene = dynamic_cast<Scene*>(g_SceneMan.GetScene());
+		GAScripted* activity = dynamic_cast<GAScripted*>(g_ActivityMan.GetActivity());
+
+		if (!scene || !activity) {
+			return false;
+		}
+
+		// Save the scene bitmaps
+		if (scene->SaveData(sc_scriptSavesPath + fileName) < 0) {
+			return false;
+		}
+
+		// We need a copy of our scene, because we have to do some fixup to remove PLACEONLOAD items and only keep the current g_movableMan state
+		Scene* sceneAltered = dynamic_cast<Scene*>(g_SceneMan.GetScene()->Clone());
+
+		// Delete any existing objects from our scene - we don't want replace broken doors or repair any stuff when we load
+		sceneAltered->ClearPlacedObjectSet(Scene::PlacedObjectSets::PLACEONLOAD, true);
+
+		// Pull all stuff from movableMan into the scene for saving, so existing actors/doors are transfered
+		sceneAltered->RetrieveSceneObjects(false); // Don't transfer ownership
```

Comment:
I'm tempted to either change this function into two seperate functions (one which transfers ownership, and one which doesn't), or otherwise using a two-valued enum with more descriptive names. (KeepOwnership, TransferOwnership?)